services/ui-src/src/components/modals/DeleteDynamicFieldModal.tsx

```diff
@@ -0,0 +1,64 @@
+// components
+import { Text } from "@chakra-ui/react";
+import { Spinner } from "@cmsgov/design-system";
+import { Modal } from "components";
+import { useState } from "react";
+// types
+import { EntityShape, EntityType } from "types";
+
+export const DeleteDynamicFieldModal = ({
+  selectedRecord,
+  removeRecord,
+  entityType,
+  modalDisclosure,
+}: Props) => {
+  const [deleting, setDeleting] = useState<boolean>(false);
+  const convertToReadableEntityName = (name: any) => {
+    const fieldTypeMap = {
+      plans: "Plan",
+      bssEntities: "BSS Entity",
+    };
+    const fieldType: keyof typeof fieldTypeMap = name;
+    return fieldTypeMap[fieldType];
+  };
+
+  const deleteProgramHandler = async () => {
+    setDeleting(true);
+    removeRecord(selectedRecord);
+    setDeleting(false);
+    modalDisclosure.onClose();
+  };
+
+  return (
+    <Modal
+      onConfirmHandler={deleteProgramHandler}
+      modalDisclosure={modalDisclosure}
+      content={{
+        heading: `Delete ${convertToReadableEntityName(entityType)}`,
```

Comment:
Wondering if this should display the entity name instead of it's general class

Reply:
I gave that a thought too but stayed with the requirements that design and the ticket gave which is why its Plan or BSS Entity instead of the specific name

Reply:
Thought: Following on my comment above about simplifying. If All of these lines that have some variant of: `${convertToReadableEntityName(entityType)}` could just reference `${fieldTypeMap[name]}` instead of that function.

Reply:
Trimmed down!